core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/database/DatabaseV4Endpoint.java

```diff
@@ -0,0 +1,95 @@
+package com.sequenceiq.cloudbreak.api.endpoint.v4.database;
+
+import java.util.Set;
+
+import javax.validation.Valid;
+import javax.validation.constraints.NotEmpty;
+import javax.ws.rs.BeanParam;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import com.sequenceiq.cloudbreak.api.endpoint.v4.database.requests.DatabaseV4ListRequest;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.database.requests.DatabaseV4Request;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.database.requests.DatabaseV4TestRequest;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.database.responses.DatabaseV4ListResponse;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.database.responses.DatabaseV4Response;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.database.responses.DatabaseV4TestResponse;
+import com.sequenceiq.cloudbreak.doc.ContentType;
+import com.sequenceiq.cloudbreak.doc.ControllerDescription;
+import com.sequenceiq.cloudbreak.doc.Notes;
+import com.sequenceiq.cloudbreak.doc.OperationDescriptions.DatabaseOpDescription;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+
+@Path("/v4/{workspaceId}/database")
+@Consumes(MediaType.APPLICATION_JSON)
+@Api(value = "/v4/{workspaceId}/database", description = ControllerDescription.DATABASES_V4_DESCRIPTION, protocols = "http,https")
+public interface DatabaseV4Endpoint {
+
+    @GET
+    @Path("")
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = DatabaseOpDescription.LIST_BY_WORKSPACE, produces = ContentType.JSON, notes = Notes.DATABASE_NOTES,
+            nickname = "listDatabasesByWorkspace")
+    DatabaseV4ListResponse list(@PathParam("workspaceId") Long workspaceId, @BeanParam DatabaseV4ListRequest databaseV4ListRequest);
```

Comment:
use Filter instead of Request in case of filter (DatabaseV4ListRequest -> DatabaseV4ListFilter)